### PR TITLE
Add Combo component

### PR DESCRIPTION
**Pull request by pratyushbanerjee**
## Description

Added new Combo component that shows a Bar and a Line chart together
Added tests and stories for Combo component

## Related Issue

https://github.com/adobe/react-spectrum-charts/issues/364

## Motivation and Context

Combo charts are useful for displaying different kinds of data in the same chart, allowing comparison between them

## How Has This Been Tested?

- Passed existing tests
- Added tests to cover changes

## Screenshots (if appropriate):

image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/stories/components/Combo/Combo.story.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+	const chartProps = useChartProps(defaultChartProps);
+	return (
+		<Chart {...chartProps}>
+			<Axis position="left" title="People" grid />
+			<Axis position="right" title="Adoption Rate" />
+			<Axis position="bottom" labelFormat="time" baseline ticks />
+			<Combo {...args}>
+				<Bar dimension="datetime" metric="people" />
+				<Line dimension="datetime" metric="adoptionRate" color={{ value: 'indigo-900' }} />
+				<Line dimension="datetime" metric="adoptionRate" color={{ value: 'indigo-900' }} />
```
Comment (marshallpete) [inline src/stories/components/Combo/Combo.story.tsx:45]: Let's just do one bar and one line for the story since that is how most people will use it.

**Comment by marshallpete** [comment]
This is looking really good. Let me investigate the pipeline to see if I can get the PR build running. It doesn't work for forks at the moment. I'll look at that tomorrow and then hopefully we can validate that it passes all the checks.

File: src/stories/components/Combo/Combo.story.tsx [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+				<Bar dimension="datetime" metric="people" />
+				<Line dimension="datetime" metric="adoptionRate" color={{ value: 'indigo-900' }} />
+			</Combo>
+		</Chart>
+	);
+};
+
+const Basic = bindWithProps(BasicComboStory);
+Basic.args = {};
+export { Basic };
```
Comment (marshallpete) [inline src/stories/components/Combo/Combo.story.tsx:52]:
```suggestion
Basic.args = {};

export { Basic };
```

File: src/stories/components/Combo/Combo.story.tsx [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+			<Combo {...args}>
+				<Bar dimension="datetime" metric="people" />
+				<Line dimension="datetime" metric="adoptionRate" color={{ value: 'indigo-900' }} />
+			</Combo>
+		</Chart>
+	);
+};
+
+const Basic = bindWithProps(BasicComboStory);
+Basic.args = {};
```
Comment (marshallpete) [inline src/stories/components/Combo/Combo.story.tsx:51]:
```suggestion
const Basic = bindWithProps(BasicComboStory);
Basic.args = {};
```

File: src/stories/components/Combo/Combo.story.tsx [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+				<Bar dimension="datetime" metric="people" />
+				<Line dimension="datetime" metric="adoptionRate" color={{ value: 'indigo-900' }} />
+			</Combo>
+		</Chart>
+	);
+};
+
+const Basic = bindWithProps(BasicComboStory);
+
+Basic.args = {};
```
Comment (marshallpete) [inline src/stories/components/Combo/Combo.story.tsx:52]:
```suggestion
const Basic = bindWithProps(BasicComboStory);
Basic.args = {};
```

File: src/stories/components/Combo/Combo.story.tsx (hunk 3 above)
Comment (marshallpete) [inline src/stories/components/Combo/Combo.story.tsx:51]:
```suggestion
const Basic = bindWithProps(BasicComboStory);

Basic.args = {};
```

**Comment by marshallpete** [comment]
It's working! Please address the failing sonarcloud issues. Let me know if you can't access them.

That took all day to figure out. It should be all good to go now. Sorry about all the tiny spacing changes. Those were to trigger the github actions.

**Comment by pratyushbanerjee** [comment]
Thanks! I have resolved the issues. But there is one code smell not related to my changes. I see it's assigned to you. Can we proceed without the fix?

**Comment by marshallpete** [comment]
Yeah no worries. I'll resolve it. Thanks! I'll finish my review in the morning.

File: src/stories/components/Combo/Combo.story.tsx [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import { Axis, Bar, Chart, Line } from '@rsc';
+import { Combo } from '@rsc/alpha';
+import { peopleAdoptionComboData } from '@stories/data/data';
+import { StoryFn } from '@storybook/react';
+import { bindWithProps } from '@test-utils';
+
+import { ChartProps } from '../../../types';
+
+export default {
+	title: 'RSC/Combo',
```
Comment (marshallpete) [inline src/stories/components/Combo/Combo.story.tsx:24]:
```suggestion
	title: 'RSC/Combo (alpha)',
```

**Review by marshallpete — APPROVED** [review]

**Pull request merged**